src/Network/NWEthernetChannel.cs

```diff
@@ -0,0 +1,208 @@
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_ethernet_channel_cancel (OS_nw_ethernet_channel ethernet_channel);
+
+		public void Cancel () => nw_ethernet_channel_cancel (GetCheckedHandle ());
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_ethernet_channel_set_queue (OS_nw_ethernet_channel ethernet_channel, IntPtr queue);
+
+		public void SetQueue (DispatchQueue queue)
+		{
+			if (queue == null)
+				throw new ArgumentNullException (nameof (queue));
+			nw_ethernet_channel_set_queue (GetCheckedHandle (), queue.Handle);
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_ethernet_channel_send (OS_nw_ethernet_channel ethernet_channel, OS_dispatch_data content, ushort vlan_tag, string remote_address, ref BlockLiteral completion);
+
+		delegate void nw_ethernet_channel_send_completion_t (IntPtr block, IntPtr error);
+		static nw_ethernet_channel_send_completion_t static_SendCompletion = TrampolineSendCompletion;
+
+		[MonoPInvokeCallback (typeof (nw_ethernet_channel_send_completion_t))]
+		static void TrampolineSendCompletion (IntPtr block, IntPtr error)
+		{
+			var del = BlockLiteral.GetTarget<Action<NWError>> (block);
+			if (del != null) {
+				var err = error == IntPtr.Zero ? null : new NWError (error, owns: false);
+				del (err);
+				err?.Dispose ();
+			}
+		}
+
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public void Send (ReadOnlySpan<byte> content, ushort vlanTag, string remoteAddress, Action<NWError> callback)
+		{
+			if (callback == null)
+				throw new ArgumentNullException (nameof (callback));
+
+			using (var dispatchData = DispatchData.FromByteBuffer (content.ToArray (), 0, content.Length)) {
+				BlockLiteral block_handler = new BlockLiteral ();
+				block_handler.SetupBlockUnsafe (static_SendCompletion, callback);
+
+				try {
+					nw_ethernet_channel_send (GetCheckedHandle (), dispatchData.GetHandle (), vlanTag, remoteAddress, ref block_handler);
+				} finally {
+					block_handler.CleanupBlock ();
+				}
+			}
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		unsafe static extern void nw_ethernet_channel_set_receive_handler (OS_nw_ethernet_channel ethernet_channel, /* [NullAllowed] */ void *handler);
+
+		delegate void nw_browser_set_state_changed_handler_t (IntPtr block, OS_dispatch_data content, ushort vlan_tag, byte[] local_address, byte[] remote_address);
+		static nw_browser_set_state_changed_handler_t static_ReceiveHandler = TrampolineReceiveHandler;
+
+		[MonoPInvokeCallback (typeof (nw_browser_set_state_changed_handler_t))]
+		static void TrampolineReceiveHandler (IntPtr block, OS_dispatch_data content, ushort vlanTag, byte[] localAddress, byte[] remoteAddress)
+		{
+			var del = BlockLiteral.GetTarget<NWEthernetChannelReceiveDelegate> (block);
+			if (del != null) {
+				DispatchData dispatchData = null, dataCopy = null;
+				IntPtr bufferAddress = IntPtr.Zero;
+				nuint bufferSize = 0;
+
+				if (content != IntPtr.Zero) {
```

Comment:
I think it would be better to use DispatchData in the delegate callback instead if ReadOnly<byte>, because it can be much faster, depending on how the callback is consumed. We could create a DispatchData.CreateMap overload that outputs a ReadOnly<byte> instance if we want to make it easier to consume.

Reply:
I have been thinking about updating DispatchData to be a more useful class. As it is atm is not very useful :/